src/System.IO.Pipelines.Text.Primitives/ReadableBufferExtensions.cs

```diff
@@ -109,14 +109,14 @@ public unsafe static ulong GetUInt64(this ReadableBuffer buffer)
                 ArraySegment<byte> data;
                 if (buffer.First.TryGetPointer(out pointer))
```

Comment:
@shiftylogic, you might want to take a look at this. It's unfortunate that this code has to fish out the ArraySegment and/or pointer out of the buffer instead of simply getting Span and parsing it. Maybe the code could simply use span and still get equal or better performance.